pkg/controllers/cloud/controller_sync_status_test.go

```diff
@@ -0,0 +1,293 @@
+			account                    *v1alpha1.CloudProviderAccount
+			selector                   *v1alpha1.CloudEntitySelector
+			secret                     *corev1.Secret
+			fakeClient                 client.WithWatch
+			pollIntv                   uint
+		)
+
+		BeforeEach(func() {
+			newScheme := runtime.NewScheme()
+			utilruntime.Must(clientgoscheme.AddToScheme(newScheme))
+			utilruntime.Must(v1alpha1.AddToScheme(newScheme))
+			utilruntime.Must(antreav1alpha2.AddToScheme(newScheme))
+			fakeClient = fake.NewClientBuilder().WithScheme(newScheme).Build()
+			pollIntv = 1
+			credentials = `{"accessKeyId": "keyId","accessKeySecret": "keySecret"}`
+			secret = &corev1.Secret{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      testSecretNamespacedName.Name,
+					Namespace: testSecretNamespacedName.Namespace,
+				},
+				Data: map[string][]byte{
+					"credentials": []byte(credentials),
+				},
+			}
+
+			account = &v1alpha1.CloudProviderAccount{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      testAccountNamespacedName.Name,
+					Namespace: testAccountNamespacedName.Namespace,
+				},
+				Spec: v1alpha1.CloudProviderAccountSpec{
+					PollIntervalInSeconds: &pollIntv,
+					AWSConfig: &v1alpha1.CloudProviderAccountAWSConfig{
+						Region: "us-east-1",
+						SecretRef: &v1alpha1.SecretReference{
+							Name:      testSecretNamespacedName.Name,
+							Namespace: testSecretNamespacedName.Namespace,
+							Key:       credentials,
+						},
+					},
+				},
+			}
+
+			selector = &v1alpha1.CloudEntitySelector{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      testSelectorNamespacedName.Name,
+					Namespace: testSelectorNamespacedName.Namespace,
+					OwnerReferences: []v1.OwnerReference{
+						{
+							APIVersion: "crd.cloud.antrea.io/v1alpha1",
+							Kind:       "CloudProviderAccount",
+							Name:       testAccountNamespacedName.Name,
+						},
+					},
+				},
+				Spec: v1alpha1.CloudEntitySelectorSpec{
+					AccountName: testAccountNamespacedName.Name,
+					VMSelector: []v1alpha1.VirtualMachineSelector{
+						{
+							VpcMatch: &v1alpha1.EntityMatch{
+								MatchID: "xyzq",
+							},
+						},
+					},
+				},
+			}
+		})
+		It("Set pending count for CPA", func() {
+			_ = fakeClient.Create(context.Background(), secret)
+			_ = fakeClient.Create(context.Background(), account)
+			var pendingSyncCount int
+			err := setPendingCount(fakeClient, &v1alpha1.CloudProviderAccountList{}, &pendingSyncCount, CPAController)
+			log().Info("setPendingCount for CPAController", "err", err)
+			Expect(err).Should(BeNil())
+			Expect(pendingSyncCount).Should(Equal(1))
+			val := IsControllerSynced(CPAController)
+			log().Info("IsControllerSynced for CPAController", "val", val)
+			Expect(val).To(BeFalse())
+			err = setPendingCount(fakeClient, &v1alpha1.VirtualMachineList{}, &pendingSyncCount, VMController)
+			log().Info("setPendingCount for CPAController", "err", err)
```

Comment:
Can we update the log to VMController?

Reply:
When VMController reconciles an VM CR, it notifies the event to converter module. Converter module calls an API server to check EE CR exists for the VM. If it does not exist, then converter module will create EE CR. 

So that is the reason, EE should reconcile all the CRs first, so that converter module does not do anything during  restart.

Reply:
Done